### PR TITLE
Tutorial framework

### PR DESCRIPTION
**Pull request by kathy-nevola**
# Title: Write a very short summary here
**Problem:** Write a description of your problem here.

**Solution:** Write a short summary of your solution.

**Key Features:**

* Key feature 1
* Key feature 2
* ...

## Checklist
- [ ] Make sure you are requesting to **pull a feature/bugfix branch** (right side). This should not be main or develop.
- [ ] Make sure you are make a pull request against either **main or develop** (left side). (Requesting to main should be reserved for bugfixes and new releases)
- [ ] Add or update unit tests (if applicable)
- [ ] Check your code with any unit tests (Run devtools::check() locally)
- [ ] Add neccessary documentation (if applicable)

## Type of changes

What type of changes does your code introduce?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue) (link the issue on the right)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Code style update (formatting, renaming)
- [ ] Documentation Update 
- [ ] Other (explain)

## Further comments
If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, any questions you have, etc...

Consider linking any issues (#issue-number ) or adding @mentions to ensure specific people see it.

### CODE REVIEW

File: OlinkAnalyze/man/figures/assay_bridgeability.png
Comment (kristynchin-olink) [inline OlinkAnalyze/man/figures/assay_bridgeability.png:1]: Examples reflect comments and feedback. Looks good.

File: OlinkAnalyze/man/figures/Bridging_schematic.png
Comment (kristynchin-olink) [inline OlinkAnalyze/man/figures/Bridging_schematic.png:1]: Schematic incorporates up-to-date terminology. Looks good.

File: OlinkAnalyze/vignettes/bridging_introduction.Rmd
Comment (kristynchin-olink) [inline OlinkAnalyze/vignettes/bridging_introduction.Rmd:1]: HT recommendation altered to reflect latest PM suggestions. New 3K-HT suggestions. Looks good, confirmed that table prints correctly in Rmd output.

File: OlinkAnalyze/vignettes/bridging_E3072toEHT.Rmd [hunk 1]
```diff
@@ -0,0 +1,682 @@
...
+section](https://cran.r-project.org/web/packages/OlinkAnalyze/vignettes/bridging_introduction.html#selecting-bridging-samples)
+of the Introduction to Bridging tutorial.
+
+## Bridge Sample Selection
+
+Prior to running a study with Explore HT, bridging samples must be selected
+from the study run with Explore 3072 and be run on the Explore HT study. These samples can
+be selected using the `olink_bridgeselector()` function in Olink Analyze
+as detailed in [the Introduction to bridging
+tutorial](https://cran.r-project.org/web/packages/OlinkAnalyze/vignettes/bridging_introduction.html#selecting-bridging-samples). The recommended number of bridge samples for within- and between- product bridging is summarized in the table below. When selecting bridge samples, the aim is to select samples that represent the dynamic range of the assay expression in the product. As such, quality control of the sample and, if available, proportion of data above LOD in the sample are considered when determining if a sample is chosen as a bridging sample. When LOD data is not available in the data expert from Olink NPX software, LOD can optionally be calculated from fixed LOD or negative controls as
```
Comment (kristynchin-olink) [inline OlinkAnalyze/vignettes/bridging_E3072toEHT.Rmd:162]: I believe "data expert" should be "data export" in line 162.

File: OlinkAnalyze/vignettes/bridging_E3072toEHT.Rmd [hunk 2]
```diff
@@ -0,0 +1,682 @@
...
+detailed in the [Calculating LOD from Olink Explore data
+tutorial](https://cran.r-project.org/web/packages/OlinkAnalyze/vignettes/LOD.html).
+
+```{r brnrtab, eval=TRUE, message=FALSE}
```
Comment (kristynchin-olink) [inline OlinkAnalyze/vignettes/bridging_E3072toEHT.Rmd:166]: This script section is printing in the vignette. Add "echo = F" to print just the table and not the corresponding script.

File: OlinkAnalyze/vignettes/bridging_E3072toEHT.Rmd [hunk 3]
```diff
@@ -0,0 +1,682 @@
...
+studies containing multiple projects of Explore 3072 data, the Explore
+3072 data sets should be bridged using within-product bridging as
+detailed in the [Introduction to bridging
+tutorial](https://cran.r-project.org/web/packages/OlinkAnalyze/vignettes/bridging_introduction.html) or otherwise normalized together
+prior to performing between-product bridging.
+
+The assays from Explore 3072 are matched to the corresponding assays in
+Explore HT and evaluated to determine if the assay is bridgeable. Additionally, all assays are normalized using both quantile smoothing and
+normalization using the median of paired differences. The result is an
+adjusted Explore 3072 data set with three additional columns:
```
Comment (kristynchin-olink) [inline OlinkAnalyze/vignettes/bridging_E3072toEHT.Rmd:202]: Should this reflect Table 4 in the Function Output section? Seems like there are 5 additional columns:

1. OlinkIDE3072 (OlinkID will reflect the HT ID)
2. MedianCenteredNPX
3. QSNormalizedNPX
4. Project
5. BridgingRecommendation

File: OlinkAnalyze/vignettes/bridging_E3072toEHT.Rmd [hunk 4]
```diff
@@ -0,0 +1,682 @@
...
+
+```{r message=FALSE, eval=FALSE, echo = TRUE}
+data_E3072 <- read_NPX("~/NPX_Explore3072_location.parquet") # Could also be a CSV file
+data_EHT <- read_NPX("~/NPX_ExploreHT_location.parquet")
+```
+
+## Checking input datasets and bridging samples
+
+First, confirm that there are overlapping sample IDs within the study.
+In a case where the same sample have different sample IDs in the two projects, the `overlap_sample_list`
```
Comment (kristynchin-olink) [inline OlinkAnalyze/vignettes/bridging_E3072toEHT.Rmd:240]: As per 9-SEP-2024 discussion, we may need to remove the line of text relating to differing overlapping sample IDs depending on what we can incorporate into the initial release. TBD on whether we will keep or remove this - for now, is OK to keep.

File: OlinkAnalyze/vignettes/bridging_E3072toEHT.Rmd [hunk 5]
```diff
@@ -0,0 +1,682 @@
...
+
+Note that external controls should not be included in the list of
+bridging samples, as detailed in the [Bridge Sample Selection] section
+of this tutorial. External control samples often share the same naming
+convention across data sets but may represent different samples due to
+reagent batch differences. Appending the project name to the end of the
+control samples can ensure unique Sample IDs.
+
+```{r, echo=TRUE, eval=FALSE}
+data.frame(SampleID = intersect(data_E3072$SampleID, data_EHT$SampleID)) |>
```
Comment (kristynchin-olink) [inline OlinkAnalyze/vignettes/bridging_E3072toEHT.Rmd:260]: Per D8493, it looks like Plate Controls, Negative Controls, Sample Controls, and Samples are output in the standard parquet format. Would it be more applicable to change the filtering criteria to dplyr::filter(SampleType == "SAMPLE")?

Comment (kristynchin-olink) [reply OlinkAnalyze/vignettes/bridging_E3072toEHT.Rmd:260]: Also not sure if it's better to use a regex for handling CSV files - example in import NPX file indicates that 3K CSV files are also valid input

Comment (kristynchin-olink) [reply OlinkAnalyze/vignettes/bridging_E3072toEHT.Rmd:260]: If we keep in str_detect, should we use the stringr::str_detect convention to match the code below? Should we add dplyr::filter and dplyr::mutate in the rbind section?

File: OlinkAnalyze/vignettes/bridging_E3072toEHT.Rmd [hunk 6]
```diff
@@ -0,0 +1,682 @@
...
+### PCA plot
+OlinkAnalyze::olink_pca_plot(df          = npx_before_br,
+                             color_g     = "Type")
+```
+
+## Normalization
+
+The `olink_normalization_bridge_product()` function is used to determine
+which assays are bridgeable and of the bridgeable assays what
+normalization method is advised, and to calculate normalized NPX values
```
Comment (kristynchin-olink) [inline OlinkAnalyze/vignettes/bridging_E3072toEHT.Rmd:306]: Might be good to split this for easier readability - possibly something like the following:

> The `olink_normalization_bridge_product()` function is used to determine which assays are bridgeable, and of the bridgeable assays, which normalization method is advised. This function also calculates normalized NPX values
> for the Explore 3072 (non-reference) project.

File: OlinkAnalyze/vignettes/bridging_E3072toEHT.Rmd [hunk 7]
```diff
@@ -0,0 +1,682 @@
...
+for the Explore 3072 (non-reference) project. Normalized NPX values are calculated for
+all assays across products as described in the Workflow Overview and in
```
Comment (kristynchin-olink) [inline OlinkAnalyze/vignettes/bridging_E3072toEHT.Rmd:308]: Would it make sense to link the Workflow Overview section? Will defer to your preference.

File: OlinkAnalyze/vignettes/bridging_E3072toEHT.Rmd [hunk 8]
```diff
@@ -0,0 +1,682 @@
...
+the sections below. Within this function the bridging recommendations
```
Comment (kristynchin-olink) [inline OlinkAnalyze/vignettes/bridging_E3072toEHT.Rmd:309]: Add comma after "within this function"

File: OlinkAnalyze/vignettes/bridging_E3072toEHT.Rmd [hunk 9]
```diff
@@ -0,0 +1,682 @@
...
+npx_br_data <- olink_normalization_bridge_product(npx_1, 
+                                   npx_2,
+                                   bridge_samples = overlap_samples_list,
+                                   exploreht_name = "reference",
+                                   explore3072_name = "new")
+```
+
+### Determining bridging recommendations
+
+For an assay to be bridgeable across products it must either have signal
```
Comment (kristynchin-olink) [inline OlinkAnalyze/vignettes/bridging_E3072toEHT.Rmd:335]: Add comma after "for an assay to be bridgeable across products"

File: OlinkAnalyze/vignettes/bridging_E3072toEHT.Rmd [hunk 10]
```diff
@@ -0,0 +1,682 @@
...
+                             "DF2" = overlap_samples)
+npx_br_data <- olink_normalization_bridge_product(npx_1, 
+                                   npx_2,
+                                   bridge_samples = overlap_samples_list,
+                                   exploreht_name = "reference",
+                                   explore3072_name = "new")
+```
+
+### Determining bridging recommendations
+
```
Comment (kristynchin-olink) [inline OlinkAnalyze/vignettes/bridging_E3072toEHT.Rmd:334]: Add comma after "for an assay to be bridgeable across products"

File: OlinkAnalyze/vignettes/bridging_E3072toEHT.Rmd [hunk 11]
```diff
@@ -0,0 +1,682 @@
...
+For an assay to be bridgeable across products it must either have signal
+in both products or be primarily background signal in both products.
+Bridging noise into signal or signal into noise can negatively impact
+downstream statistical analysis. To determine if an assay is bridgeable,
+the following criteria are considered:
+
```
[the remark this answers is not in the document]
Comment (kristynchin-olink) [reply OlinkAnalyze/vignettes/bridging_E3072toEHT.Rmd:340]: Line 352 does not have the superscript in R^2^ like the others do

File: OlinkAnalyze/vignettes/bridging_E3072toEHT.Rmd [hunk 12]
```diff
@@ -0,0 +1,682 @@
...
+        for bridging, the median number of counts in both products is
+        calculated, excluding data points with less than 10 counts. If
+        the median number of counts is less than the cutoff then the
+        assay does not have sufficient counts to be used for bridging.
+        The default cutoff is set to 150 counts, which is based on the
+        count quality control metrics for Explore products.
+
+For assays that are bridgeable, the shape of the NPX distribution is
+compared between the two products:
+
```
Comment (kristynchin-olink) [inline OlinkAnalyze/vignettes/bridging_E3072toEHT.Rmd:383]: In line 394, add a comma after "If the KS statistic is above the cutoff"

File: OlinkAnalyze/vignettes/bridging_E3072toEHT.Rmd [hunk 13]
```diff
@@ -0,0 +1,682 @@
...
+    differences is recommended. By default this cutoff difference is set
+    to 0.2.
+
+An overview of these criteria is visualized below.
+
+```{r, fig.cap= fcap, eval=TRUE, echo=FALSE, out.width="100%"}
+knitr::include_graphics(normalizePath("../man/figures/assay_bridgeability.png"), error = FALSE)
+fcap <- "Criteria to determine the bridging recommendation for an assay. The assessment of linearity ensures bridging between signal in both platforms or noise in both platforms (but not between signal and noise. Similar NPX ranges and sufficient counts provide additional insight into an assays bridgeability. Distribution shape is assessed to determine recommended bridging method."
+```
+
```
Comment (kristynchin-olink) [inline OlinkAnalyze/vignettes/bridging_E3072toEHT.Rmd:407]: Optional, but is there a way to add some more space between the caption and the text below? Given the length of the caption and same formatting as the regular text, at first glance it looks like the caption is a continuation of the text normal text.

Comment (kristynchin-olink) [reply OlinkAnalyze/vignettes/bridging_E3072toEHT.Rmd:407]: In line 405, there is a missing closing parenthesis after "(but not between signal and noise"

Comment (kristynchin-olink) [reply OlinkAnalyze/vignettes/bridging_E3072toEHT.Rmd:407]: In line 405, change "assays" to "assay's"

File: OlinkAnalyze/vignettes/bridging_E3072toEHT.Rmd [hunk 14]
```diff
@@ -0,0 +1,682 @@
...
+    the NPX values of the bridging samples for an assay, to create a step
+    function which interpolates linearly between the available data
+    points.
+
+2.  The empirical distribution function is used to map the data points
+    from Explore 3072 to the Explore HT space using the specified
+    quantiles. At this point all data points from the bridging samples
+    have NPX values that are normalized to the data points in Explore
+    HT.
+
```
Comment (kristynchin-olink) [inline OlinkAnalyze/vignettes/bridging_E3072toEHT.Rmd:465]: Add comma on line 466 after "to normalize the remaining data"

Comment (kristynchin-olink) [reply OlinkAnalyze/vignettes/bridging_E3072toEHT.Rmd:465]: Confirming that Explore 3072 should be referenced twice in the bolded section: 

> To normalize the remaining data a spline regression model is constructed **using the sorted Explore 3072 data and the mapped Explore 3072 data**

If this refers to the Explore 3072 data only, would it be clearer to say "using the sorted and mapped Explore 3072 data?

File: OlinkAnalyze/vignettes/bridging_E3072toEHT.Rmd (hunk 6 above)
Comment (kristynchin-olink) [reply OlinkAnalyze/vignettes/bridging_E3072toEHT.Rmd:306]: Also TBD - will we have the olink_normalization_bridge_product() wrapper ready for initial launch? If not, should we just refer to olink_normalization()? Will leave as-is for now.

File: OlinkAnalyze/vignettes/bridging_E3072toEHT.Rmd [hunk 15]
```diff
@@ -0,0 +1,682 @@
...
+  mutate(Project = "data1") 
+npx_2 <- data_E3072 |>
+  mutate(Project = "data2")
+
+overlap_samples <-data.frame(SampleID = intersect(npx_1$SampleID, npx_2$SampleID)) |>
+  filter(!str_detect(SampleID, "CONTROL_SAMPLE")) |> #Remove control samples
+  pull(SampleID)
+
+overlap_samples_list <- list("DF1" = overlap_samples,
+                             "DF2" = overlap_samples)
```
Comment (kristynchin-olink) [inline OlinkAnalyze/vignettes/bridging_E3072toEHT.Rmd:325]: TBD - will we have the olink_normalization_bridge_product() wrapper ready for initial launch? If not, should we just refer to olink_normalization()? Will leave as-is for now.

File: OlinkAnalyze/vignettes/bridging_E3072toEHT.Rmd [hunk 16]
```diff
@@ -0,0 +1,682 @@
...
+
+4.  The spline regression model is then used to predict all the data
+    points from Explore 3072 to Explore HT. The spline regression model
+    results in a combination of linear regression models within
+    intervals. The Explore 3072 NPX values are input as the x value
+    within the corresponding interval, which results in a y value
+    equivalent to the Explore HT NPX value.
+
+### Function Output
+
```
Comment (kristynchin-olink) [inline OlinkAnalyze/vignettes/bridging_E3072toEHT.Rmd:481]: TBD - will we have the olink_normalization_bridge_product() wrapper ready for initial launch? If not, should we just refer to olink_normalization()? Will leave as-is for now.

File: OlinkAnalyze/vignettes/bridging_E3072toEHT.Rmd [hunk 17]
```diff
@@ -0,0 +1,682 @@
...
+run.
+
+Within each product, the assays contain unique OlinkID values for each
+of their corresponding panels and blocks in Explore 3072 and Explore HT,
+respectively. It should be noted that the overlapping assays themselves
+differ between each product and that these assays differ from the
+overlapping assays *across* products.
+
+### Downstream Analysis
+
```
Comment (kristynchin-olink) [inline OlinkAnalyze/vignettes/bridging_E3072toEHT.Rmd:616]: TBD - will we have the olink_normalization_bridge_product() wrapper ready for initial launch? If not, should we just refer to olink_normalization()? Will leave as-is for now.

File: OlinkAnalyze/vignettes/bridging_E3072toEHT.Rmd [hunk 18]
```diff
@@ -0,0 +1,682 @@
...
+lists what method, if any should be used for that assay. Additional
+columns including `OlinkID_HT` and `OlinkID_3072` map the assays across
+products and the `Project` column lists the name of the project based on
+the `exploreht_name` and `explore3072_name` arguments. The resulting
+data set will contain the newly bridged Explore 3072 data set. The
+reference Explore HT data will be concatenated to the Explore 3072 data.
+As the reference data is not altered during normalization, the
+normalized NPX values in the Explore HT data will be the same as the
+values in the NPX column which contains the non-normalized data.
+
```
Comment (kristynchin-olink) [inline OlinkAnalyze/vignettes/bridging_E3072toEHT.Rmd:497]: Should there be another column in the example table below called OlinkID_HT as described on line 489? Should OlinkID be renamed as OlinkID_HT? Or should there be 3 separate OID columns?

File: OlinkAnalyze/vignettes/bridging_E3072toEHT.Rmd (hunk 3 above)
Comment (kristynchin-olink) [reply OlinkAnalyze/vignettes/bridging_E3072toEHT.Rmd:202]: Function Output section also mentions an OlinkID_HT column that's added to the output.

File: OlinkAnalyze/vignettes/bridging_E3072toEHT.Rmd (hunk 18 above)
Comment (kristynchin-olink) [reply OlinkAnalyze/vignettes/bridging_E3072toEHT.Rmd:497]: Table shows "MedianCenteredNPX" but text says that this is named "MedianAdjustedNPX" - which one is accurate?

File: OlinkAnalyze/vignettes/bridging_E3072toEHT.Rmd [hunk 19]
```diff
@@ -0,0 +1,682 @@
...
+PCA is used to assess the quality of bridging by determining if the
+sample controls (SCs) and bridging samples appear closer after bridging.
+Two PCAs can be generated, one containing the SCs and one containing the
+bridging samples. Prior to bridging there will be a noticeable
+separation between products which should decrease after bridging.
+
+```{r pca2, message=FALSE, fig.cap= f8, eval=FALSE, echo = TRUE}
+## After bridging
+
```
Comment (kristynchin-olink) [inline OlinkAnalyze/vignettes/bridging_E3072toEHT.Rmd:545]: Should all filter and mutate steps include "dplyr::" for consistency with other lines?

File: OlinkAnalyze/vignettes/bridging_E3072toEHT.Rmd [hunk 20]
```diff
@@ -0,0 +1,682 @@
...
+blocks. These assays are included for QC purposes and allow users to
+evaluate data performance across panels in Explore 3072 and across
+blocks in Explore HT. IL6, IL8 (CXCL8), and TNF are included in the
+Cardiometabolic, Oncology, Neurology and Inflammation panels, while
+IDO1, LMOD1, and SCRIB are included in the Cardiometabolic II, Oncology
+II, Neurology II and Inflammation II panels. Each correlation assay is
+measured four times in an Olink Explore 3072 run. In Explore HT, GBP1
+and MAPK1 serve as correlation assays and are measured three times in a
+run.
+
```
Comment (kristynchin-olink) [inline OlinkAnalyze/vignettes/bridging_E3072toEHT.Rmd:608]: I think the last sentence possibly repeats the same thing twice - should we cut down 

> It should be noted that the overlapping assays themselves differ between each product and that these assays differ from the overlapping assays *across* products.

to something like the following?

It should be noted that the 

> overlapping assays differ *across* products.

I also may just not be understanding! I will defer to you.

File: OlinkAnalyze/vignettes/bridging_E3072toEHT.Rmd
Comment (kristynchin-olink) [inline OlinkAnalyze/vignettes/bridging_E3072toEHT.Rmd:1]: Confirm that all links work as expected. Some minor comments and double checks left throughout Rmd.

**Review by kristynchin-olink — CHANGES_REQUESTED** [review]
Some small changes requested in 3072 to HT bridging vignette.

File: OlinkAnalyze/vignettes/bridging_E3072toEHT.Rmd (hunk 5 above)
Comment (kathy-nevola) [reply OlinkAnalyze/vignettes/bridging_E3072toEHT.Rmd:260]: I changed all to SampleType (and AssayType) and added a note if these columns arent present.

File: OlinkAnalyze/vignettes/bridging_E3072toEHT.Rmd (hunk 14 above)
Comment (kathy-nevola) [reply OlinkAnalyze/vignettes/bridging_E3072toEHT.Rmd:465]: Ive reworded this to be clearer: To normalize the remaining data, a spline regression model is
    constructed using the sorted Explore 3072 data (prior to mapping) and the mapped
    Explore 3072 data, along with the anchor points of the spline
    function.

File: OlinkAnalyze/vignettes/bridging_E3072toEHT.Rmd
Comment (kristynchin-olink) [reply OlinkAnalyze/vignettes/bridging_E3072toEHT.Rmd:1]: All requested changes addressed. Vignette looks great.

**Review by kristynchin-olink — APPROVED** [review]

**Comment by amrita-kar** [comment]
- Might not be clear to the client in the "Are there sufficient counts in "=assay_bridgeability.png why the second case is marked not possible. Do we need to clarify?

**Review by amrita-kar — APPROVED** [review]

**Pull request merged**